src/clj_3df/compiler.cljc

```diff
@@ -222,20 +224,24 @@
           {:Filter [args (encode-predicate predicate) :_ offset->const]}
           (throw (ex-info "All predicate inputs must be bound in a single relation." {:binding this})))))))
 
-(defrecord Aggregation [fn-symbols args key-symbols binding symbols]
+(defrecord Aggregation [fn-symbols args key-symbols binding symbols with]
   IBinding
   (bound-symbols [this] symbols)
   (plan [this]
-    (let [symbols (bound-symbols this)]
+    (let [symbols      (bound-symbols this)
+          with-symbols (or with [])]
       (if (binds-all? binding symbols)
-        {:Aggregate [symbols (plan binding) (map (comp str/upper-case name) fn-symbols) key-symbols args]}
+        {:Aggregate [symbols (plan binding) (map (comp str/upper-case name) fn-symbols) key-symbols args with-symbols]}
         (if debug?
-          {:Aggregate [args :_ (map (comp str/upper-case name) fn-symbols) symbols]}
+          {:Aggregate [args :_ (map (comp str/upper-case name) fn-symbols) symbols with-symbols]}
           (throw (ex-info "Aggregation on unbound symbols." {:binding (debug-plan this)})))))))
 
 (defrecord Projection [binding symbols]
   IBinding
-  (bound-symbols [this] symbols)
+  (bound-symbols [this]
+    (if (apply distinct? symbols)
```

Comment:
As seen below we concat the `:with` symbols into the distinct `:find `symbols. 
If the user requested the same symbols in the with and find clause we throw, as then there are duplicates in the `symbols`

Reply:
It would be a bit more tricky in the backend if we would allow duplicated `:find` and `:with` symbols, besides it making semantically no sense.
Or what is your take on it?